python/run-tests.py

```diff
@@ -107,20 +122,26 @@ def run_individual_python_test(target_dir, test_name, pyspark_python):
     env["PYSPARK_SUBMIT_ARGS"] = " ".join(spark_args)
 
     output_prefix = get_valid_filename(pyspark_python + "__" + test_name + "__").lstrip("_")
-    per_test_output = tempfile.NamedTemporaryFile(prefix=output_prefix, suffix=".log")
+
+    if keep_test_output:
+        # The location is unique because the test is already in a unique directory.
+        per_test_output = open(os.path.join(tmp_dir, output_prefix + ".log"), "wb+")
+    else:
+        per_test_output = tempfile.NamedTemporaryFile(prefix=output_prefix, suffix=".log")
```

Comment:
> As for this, my proposal here would be to reuse the existing target dir option and simply move the log output into this path.

If we want to keep the option as a flag, I believe we can pass `delete=False`.

Reply:
Ok, will do